tests/nnfw_api/src/ModelTestDynamicTensor.cc

```diff
@@ -22,31 +22,89 @@
 #include "CircleGen.h"
 #include "GenModelTest.h"
 
-void set_input_output(nnfw_session *session, const std::vector<float> &input,
-                      std::vector<float> &actual_output)
+// This macro can be used instead of using NNFW_ENSURE_SUCCESS especially with negative test.
+// E.g., setInputOutput() is written with this macro and the following check is available to check
+// if there's any error while setting input or output:
+//
+//  EXPECT_ANY_THROW(setInputOutput(...));
+//
+#define THROW_WHEN_NNFW_ERROR(result)                                  \
+  do                                                                   \
+  {                                                                    \
+    if (result != NNFW_STATUS_NO_ERROR)                                \
+      throw std::runtime_error("returning error on calling nnfw api"); \
+  } while (false);
```

Comment:
I would like to suggest 

```suggestion
  } while (false)
```

I think it is more common used without `;`. (to force developers always put `;` when they use this macro)

Reply:
super!